ldm/invoke/generator/inpaint.py

```diff
@@ -31,17 +35,21 @@ def get_make_image(self,prompt,sampler,steps,cfg_scale,ddim_eta,
         the time you call it.  kwargs are 'init_latent' and 'strength'
         """
 
-        # Get the alpha channel of the mask
-        pil_init_mask = pil_mask.getchannel('A')
-        pil_init_image = pil_image.convert('RGBA') # Add an alpha channel if one doesn't exist
-
-        # Build an image with only visible pixels from source to use as reference for color-matching.
-        # Note that this doesn't use the mask, which would exclude some source image pixels from the
-        # histogram and cause slight color changes.
-        init_rgb_pixels = np.asarray(pil_image.convert('RGB'), dtype=np.uint8).reshape(pil_image.width * pil_image.height, 3)
-        init_a_pixels = np.asarray(pil_init_image.getchannel('A'), dtype=np.uint8).reshape(pil_init_mask.width * pil_init_mask.height)
-        init_rgb_pixels = init_rgb_pixels[init_a_pixels > 0]
-        init_rgb_pixels = init_rgb_pixels.reshape(1, init_rgb_pixels.shape[0], init_rgb_pixels.shape[1]) # Filter to just pixels that have any alpha, this is now our histogram
+        if isinstance(init_image, PIL.Image.Image):
```

Comment:
If `init_image` isn't a Pillow image, we won't set `pil_image`, and `sample_to_image` below will break. Should we either assert it's a Pillow image here, or assume it is (possibly adding a type hint to the function definition) and let it fail if not?

Reply:
Same issue with `mask_image` right below this

Reply:
Good point. In my version this case would just fall back to the previous behaviour and not try the post processing. I’ll adjust to do the same.

Reply:
I've fixed. In the event that a tensor (or something else) is passed, the post-generation fixup code will not execute and behavior will be as before.